test/WebSocketSessionTest.cpp

```diff
@@ -0,0 +1,159 @@
+//   Copyright 2023, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include <gtest/gtest.h>
+
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/beast/core/flat_buffer.hpp>
+
+#include "util/AsyncTestHelpers.h"
+#include "util/BeastTestHelpers.h"
+#include "util/http/websocket/WebSocketSession.h"
+
+using ad_utility::websocket::QueryHub;
+using ad_utility::websocket::QueryId;
+using ad_utility::websocket::WebSocketSession;
+namespace net = boost::asio;
+namespace http = boost::beast::http;
+using net::ip::tcp;
+using namespace boost::asio::experimental::awaitable_operators;
+
+http::request<http::string_body> withPath(std::string_view path) {
+  http::request<http::string_body> request{};
+  request.target(path);
+  return request;
+}
+
+TEST(WebSocketSession, EnsureCorrectPathAcceptAndRejectBehaviour) {
+  auto testFunc = WebSocketSession::getErrorResponseIfPathIsInvalid;
+
+  EXPECT_EQ(std::nullopt, testFunc(withPath("/watch/some-id")));
+  EXPECT_EQ(std::nullopt, testFunc(withPath("/watch/😇")));
+
+  auto assertNotFoundResponse = [](auto response) {
+    ASSERT_TRUE(response.has_value());
+    EXPECT_EQ(response->result(), http::status::not_found);
+  };
+
+  assertNotFoundResponse(testFunc(withPath("")));
+  assertNotFoundResponse(testFunc(withPath("/")));
+  assertNotFoundResponse(testFunc(withPath("/watch")));
+  assertNotFoundResponse(testFunc(withPath("/watch/")));
+  assertNotFoundResponse(testFunc(withPath("/watch//")));
+  assertNotFoundResponse(testFunc(withPath("/watch///")));
+  assertNotFoundResponse(testFunc(withPath("/watch/trailing-slash/")));
+  assertNotFoundResponse(testFunc(withPath("/other-prefix/some-id")));
+}
+
+// Hack to allow ASSERT_*() macros to work with ASYNC_TEST
+#define return co_return
+
+ASYNC_TEST(WebSocketSession, verifySessionEndsOnClientCloseWhileTransmitting) {
```

Comment:
I don't quite understand the name. Why `while transmitting`?

Reply:
While updates are transmitted from the server to the client, before they are completed

Reply:
In other words it tests that the server handles an abrupt socket close gracefully